Sources/Instrumentation/Tracing/TracingInstrument.swift

```diff
@@ -0,0 +1,27 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Dispatch
+
+public protocol TracingInstrument: Instrument {
+    var currentSpan: Span? { get }
+
+    func startSpan(named operationName: String, baggage: BaggageContext, at timestamp: DispatchTime) -> Span
+}
+
```

Comment:
are we sure we need the explicit `at timestamp: DispatchTime` param?

Reply:
I thought it might be helpful to have an option to supply a timestamp when testing. There's also a write up about this in the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#span-creation

Reply:
👍

Reply:
Should we express it as `DispatchTime?` so it's clearer why context is not last?

Reply:
Yes, I like that 👍

Reply:
Cool excellent